### PR TITLE
Update hsplit in np_array_ops.py

### PR DESCRIPTION
**Pull request by xoiga123**
[numpy docs](https://numpy.org/doc/stable/reference/generated/numpy.hsplit.html): 
> hsplit is equivalent to split with axis=1, the array is always split along the second axis except for 1-D arrays, where it is split at axis=0.

As there is no testVsplit or testHsplit in `np_array_ops_test.py`, this fix doesn't need any test modification (and prob why the bug exists)

### CODE REVIEW

File: tensorflow/python/ops/numpy_ops/np_array_ops.py [hunk 1]
```diff
@@ -1015,6 +1015,9 @@ def _split_on_axis(np_fun_name, axis):
 
   @np_utils.np_doc(np_fun_name)
   def f(ary, indices_or_sections):
+    # for 1-D array, hsplit becomes vsplit
```
Comment (JW1992) [inline tensorflow/python/ops/numpy_ops/np_array_ops.py:1018]: Thanks for the PR! Do you mind adding a test as well? Like when the input is a 1-D array

**Review by JW1992 — CHANGES_REQUESTED** [review]
Requesting two changes:
1. Replace the static shape branches into dynamic branches so that the shape information is processed at graph run time;
2. Add a test to ensure that the changes in 1 are covered

**Comment by xoiga123** [comment]
@JW1992 I passed the test you wrote, but may I ask where to put that test in? I don't think `np_array_ops_test.py` has any dynamic shape test.

File: tensorflow/python/ops/numpy_ops/np_array_ops.py [hunk 2]
```diff
@@ -1017,8 +1017,15 @@ def _split_on_axis(np_fun_name, axis):
...
-    if axis == 1 and len(ary.shape) == 1:
-      axis = 0
+    axis = np_utils.cond(
+      math_ops.equal(axis, 1),
+      lambda: np_utils.cond(
+        math_ops.equal(array_ops.rank(ary), 1),
+        lambda: 0,
+        lambda: 1
+      ),
+      lambda: 1
```
Comment (JW1992) [inline tensorflow/python/ops/numpy_ops/np_array_ops.py:1027]: Does this take care of the case when `axis=2` (i.e. dsplit)?

File: tensorflow/python/ops/numpy_ops/np_array_ops.py [hunk 3]
```diff
@@ -1017,8 +1017,15 @@ def _split_on_axis(np_fun_name, axis):
...
+    )
```
Comment (JW1992) [inline tensorflow/python/ops/numpy_ops/np_array_ops.py:1028]: For the test can you please add it to tensorflow/python/ops/numpy_ops/np_array_ops_test.py? Thanks

**Review by JW1992 — CHANGES_REQUESTED** [review]
Thanks!

**Comment by xoiga123** [comment]
I wrote `testHVDSplit` like [jax's implementation](https://github.com/google/jax/blob/f337c00ed517d1ab012ac204d9974e167b33550c/tests/lax_numpy_test.py#L2757)

File: tensorflow/python/ops/numpy_ops/np_array_ops_test.py [hunk 4]
```diff
@@ -1156,6 +1156,27 @@ def testSplit(self):
...
+    def f(arr, axis):
+      if axis == 0:
+        return np_array_ops.vsplit(arr, 2)
+      elif axis == 1:
+        return np_array_ops.hsplit(arr, 2)
+      else:
+        return np_array_ops.dsplit(arr, 2)
+
+    x = np_array_ops.arange(4)
+    self.assertListEqual([([0, 1]), ([2, 3])], f(x, constant_op.constant(0)))
```
Comment (JW1992) [inline tensorflow/python/ops/numpy_ops/np_array_ops_test.py:1174]: QQ: Are these extra tuples `()` necessary?

**Review by JW1992 — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/numpy_ops/np_array_ops.py [hunk 5]
```diff
@@ -1014,13 +1014,23 @@ def split(ary, indices_or_sections, axis=0):
 def _split_on_axis(np_fun_name, axis):
 
   @np_utils.np_doc(np_fun_name)
-  def f(ary, indices_or_sections):
+    def f(ary, indices_or_sections):
```
[the remark this answers is not in the document]
Comment (xoiga123) [reply tensorflow/python/ops/numpy_ops/np_array_ops.py:1017]: my bad :cry:

File: tensorflow/python/ops/numpy_ops/np_array_ops_test.py (hunk 4 above)
Comment (xoiga123) [reply tensorflow/python/ops/numpy_ops/np_array_ops_test.py:1174]: I followed `testSplit` convention, so didn't really know. Waiting for bazel to build the test.

Btw may I ask when does `bazel` caches a build? I built this morning and with incremental changes, it reused a cache and was really fast. After a nap, running `bazel test` with a minor change (remove these `()`) rebuilds the whole thing again, which takes ~2hrs

**Review by JW1992 — APPROVED** [review]

**Pull request merged**